src/main/java/com/napier/sem/DAO.java

```diff
@@ -77,7 +78,7 @@ public static String getWhereCondition(String areaFilter, String areaName)
             // Create Country object and add it to the list for each result in the query
             while (resultSet.next())
             {
-                records.add(new Record(resultSet, recordType));
+                if (resultSet.getString("name") != null) { records.add(new Record(resultSet, recordType)); }
```

Comment:
Curly brackets should be moved on to their own lines to keep with the formatting in the rest of the file